pilot/pkg/security/model/authentication_test.go

```diff
@@ -358,8 +425,233 @@ func TestApplyToCommonTLSContext(t *testing.T) {
 			tlsContext := &auth.CommonTlsContext{}
 			ApplyToCommonTLSContext(tlsContext, test.node.Metadata, test.sdsUdsPath, []string{})
 
-			if !reflect.DeepEqual(tlsContext, test.result) {
-				t.Errorf("got() = %v, want %v", spew.Sdump(tlsContext), spew.Sdump(test.result))
+			if !cmp.Equal(tlsContext, test.result, protocmp.Transform()) {
+				t.Errorf("got(%#v), want(%#v)\n", spew.Sdump(tlsContext), spew.Sdump(test.result))
+			}
+		})
+	}
+}
+
+func TestApplyCustomSDSToCommonTLSContext(t *testing.T) {
+	testCases := []struct {
+		name       string
+		sdsUdsPath string
+		tlsOpts    *networking.ServerTLSSettings
+		result     *auth.CommonTlsContext
+	}{
+		{
+			name:       "static certificate validation without SubjectAltNames ",
+			sdsUdsPath: "/tmp/sdsuds.sock",
+			tlsOpts: &networking.ServerTLSSettings{
+				CredentialName: "spiffe://cluster.local/ns/bar/sa/foo",
+				Mode:           networking.ServerTLSSettings_SIMPLE,
+			},
+			result: &auth.CommonTlsContext{
+				TlsCertificateSdsSecretConfigs: []*auth.SdsSecretConfig{
+					{
+						Name: "spiffe://cluster.local/ns/bar/sa/foo",
+						SdsConfig: &core.ConfigSource{
+							InitialFetchTimeout: features.InitialFetchTimeout,
+							ConfigSourceSpecifier: &core.ConfigSource_ApiConfigSource{
+								ApiConfigSource: &core.ApiConfigSource{
+									ApiType: core.ApiConfigSource_GRPC,
+									GrpcServices: []*core.GrpcService{
+										{
+											TargetSpecifier: &core.GrpcService_GoogleGrpc_{
+												GoogleGrpc: &core.GrpcService_GoogleGrpc{
+													TargetUri:  "/tmp/sdsuds.sock",
+													StatPrefix: SDSStatPrefix,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:       "static certificate validation with SubjectAltNames ",
+			sdsUdsPath: "/tmp/sdsuds.sock",
+			tlsOpts: &networking.ServerTLSSettings{
+				CredentialName:  "spiffe://cluster.local/ns/bar/sa/foo",
+				Mode:            networking.ServerTLSSettings_PASSTHROUGH,
+				SubjectAltNames: []string{"SystemCACertificates.keychain", "SystemRootCertificates.keychain"},
```

Comment:
Use different names in the SubjectAltNames so that you can make sure all of them are used in the ValidationContextType.